bin/scanpy_wrapper_utils.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+import sys
+import logging
+import argparse
+import scanpy as sc
+
+
+def comma_separated_list(name, dtyp):
+    supported_dtypes = {'integer':int, 'numeric':float, 'string':str}
+    if dtyp not in supported_dtypes:
+        logging.warn('unsupported data type: "{}", default to "string"'.format(dtyp))
+        dtype_name = 'string'
+        dtype = str
+    else:
+        dtype_name = dtyp
+        dtype = supported_dtypes[dtyp]
+
+    def parse_comma_separated_list(arg, arg_name=name, dtype=dtype, dtype_name=dtype_name):
+        L = arg.split(',')
+        V = []
+        for s in L:
+            try:
+                v = dtype(s)
+            except ValueError:
+                logging.error('--{} expects comma separated list of [{}] but received "{}"'.format(arg_name, dtype_name, s))
+                break
+            V.append(v)
+        return V
+
+    return parse_comma_separated_list
+
+
+class ScanpyArgParser(object):
+    def __init__(self, description=None):
+        self.parser = argparse.ArgumentParser(description=description)
+        self.parser.add_argument('--debug', action='store_true',
+                                 help='Print debug information.')
+        self.names = set()
+
+
+    def add_input_object(self):
+        self.parser.add_argument('-i', '--input-object-file',
+                                 required=True,
+                                 help='Path to anndata or loom file.')
+        self.parser.add_argument('-f', '--input-format',
+                                 choices=['loom', 'anndata', 'auto-detect'],
+                                 default='auto-detect',
+                                 help='Format for input object: loom/anndata/[auto-detect].')
+        self.names.add('input-object-file')
+        self.names.add('input-format')
+
+
+    def add_output_object(self):
+        self.parser.add_argument('-o', '--output-object-file',
+                                 required=True,
+                                 help='File name in which to store serialized python object.')
+        self.parser.add_argument('-F', '--output-format',
+                                 choices=['loom', 'anndata', 'auto-detect'],
+                                 default='auto-detect',
+                                 help='Format for output object: loom/anndata/[auto-detect].')
+        self.names.add('output-object-file')
+        self.names.add('output-format')
+
+
+    def add_subset_parameters(self):
```

Comment:
I think that this sort of very specific parameter handling should be part of each of the functional that uses it, instead of being here. If you include very specific functions here you'll end a having conflicts between branches often.